### ISSUE TITLE
how to use quasar directives in nicegui?

### ISSUE THREAD

**Issue by CrystalWindSnake** [zauberzeug/nicegui#1199]
### Description

I hope to use the convenient directives provided by quasar in a nicegui component.

such as close-popup directives 。this is link [quasar close-popup](https://quasar.dev/vue-directives/close-popup)

However, it does not work when used in nicegui.

```python
from nicegui import ui

with ui.dialog(value=True).props("persistent"):
    ui.button("test").props("v-close-popup")

ui.run()
```

**Comment by falkoschindler** [comment]
Interesting, @CrystalWindSnake!

There are several things going on:

1. Without placing the buttons inside some container like `ui.card` within the dialog, the buttons are not _really_ clickable. At least the cursor doesn't indicate it. And without the card event `d.close` doesn't work in combination with the "persistent" prop.
2. The "persistent" prop seems to block "v-close-popup". Apparently closing a dialog using this directive is treated like dismissing it via escape or clicking in the background, which is forbidding with persistent dialogs.
3. Without the "persistent" prop and with a card container, the directive works equally well like the close method:
    ```py
    with ui.dialog(value=True).props('persistent') as d
        with ui.card():
            ui.button("Close").props("v-close-popup")
            ui.button("Close", on_click=d.close)
    ```

Long story short: `v-close-popup` basically works. I'm just not sure if I would personally prefer it over calling `d.close` explicitly. But that's up to you. 🙂

**Comment by CrystalWindSnake** [comment]
@falkoschindler That's true, thank you for your detailed explanation.
